pkg/service/service.go

```diff
@@ -1596,11 +1604,14 @@ func (s *Service) deleteServiceLocked(svc *svcInfo) error {
 	delete(s.svcByHash, svc.hash)
 	delete(s.svcByID, svc.frontend.ID)
 
-	for _, id := range obsoleteBackendIDs {
-		scopedLog.WithField(logfields.BackendID, id).
-			Debug("Deleting obsolete backend")
-		s.lbmap.DeleteBackendByID(id)
+	if !headless {
+		for _, id := range obsoleteBackendIDs {
+			scopedLog.WithField(logfields.BackendID, id).
+				Debug("Deleting obsolete backend")
+			s.lbmap.DeleteBackendByID(id)
+		}
 	}
+
```

Comment:
There's a bit of asymmetry in the operations below, since both the metric increase and the emission of the notification is performed in case of headless service deletion, but not creation/update.